acceptance/analyzer_test.go

```diff
@@ -868,6 +874,27 @@ func testAnalyzerFunc(platformAPI string) func(t *testing.T, when spec.G, it spe
 						})
 					})
 				})
+
+				when("do have read access to registry", func() {
+					it("throw read error", func() {
+						h.SkipIf(t, api.MustParse(platformAPI).Compare(api.MustParse("0.7")) < 0, "Platform API < 0.7 does not use tag flag")
+						cmd := exec.Command(
+							"docker", "run", "--rm",
+							"--network", registryNetwork,
+							"--env", "CNB_PLATFORM_API="+platformAPI,
+							"--name", containerName,
+							analyzeImage,
+							ctrPath(analyzerPath),
+							"-previous-image", "my-previous-image",
+							"some-image",
```

Comment:
I'm curious why this is throwing a read error - shouldn't the analyzer proceed happily if it can't find the previous image? What about `my-previous-image` and `some-image` makes them not readable?

Reply:
as you can see the -image-name and -previous-image are actually not pointing to the registry, in this case it is using docker.io as default host url and it is actually returning a 403 respond. basically that's why it has `no read` access

Reply:
Ohh I see... so that must mean there is some image on docker.io called "some-image" that we don't have access to? I would rather not be dependent on the state of Docker Hub for these tests to work (what if someone makes that image public? It's unlikely to happen, but still...) - could we make them images in the test auth registry?

Reply:
@natalieparellano I created the following [PR](https://github.com/buildpacks/imgutil/pull/125) in imgutil to help me to solve the issue. The main problem is our current implementation of the dockerRegistry only have two behaviors: **read only** or **basic auth,** in this case `some-image` requires read/write access and `previous-image` requires read access, I can't handle these behaviors. 

I implemented a new custom handler in the docker registry to do what I need, it is not perfect, I think we can refactor to remove duplicated code but I would like you to take a look and give me your thoughts. Using the changes from the imgutil I can add the following code to this test.

```
when("no read access", func() {
					var customRegistry *ih.DockerRegistry
					var previousImage string

					it.Before(func() {
						customDockerConfigDir, err := ioutil.TempDir("", "test.docker.config.custom.dir")
						h.AssertNil(t, err)
						defer os.RemoveAll(customDockerConfigDir)

						var customPrivileges = make(map[string]ih.ImagePrivileges)
						customPrivileges["some-image"] = ih.ImagePrivileges{Readable: true, Writeable: true}
						customPrivileges["my-previous-image"] = ih.ImagePrivileges{Readable: false, Writeable: false}

						sharedRegHandler := registry.New(registry.Logger(log.New(ioutil.Discard, "", log.Lshortfile)))
						customRegistry = ih.NewDockerRegistry(ih.WithAuth(customDockerConfigDir), ih.WithSharedHandler(sharedRegHandler),
							ih.WithCustomPrivileges(customPrivileges))

						customRegistry.Start(t)
						defer customRegistry.Stop(t)

						imageName = customRegistry.RepoName("some-image")
						authConfig, err = auth.BuildEnvVar(authn.DefaultKeychain, imageName)
						previousImage = customRegistry.RepoName("my-previous-image")
						h.AssertNil(t, err)
					})

					it("throws read error accessing previous image", func() {
						cmd := exec.Command(
							"docker", "run", "--rm",
							"--network", registryNetwork,
							"--env", "CNB_PLATFORM_API="+platformAPI,
							"--env", "CNB_REGISTRY_AUTH="+authConfig,
							"--name", containerName,
							analyzeImage,
							ctrPath(analyzerPath),
							"-previous-image", previousImage,
							imageName,
						) // #nosec G204
						output, err := cmd.CombinedOutput()

						h.AssertNotNil(t, err)
						expected := "failed to : ensure registry read access to " + previousImage
						h.AssertStringContains(t, string(output), expected)
					})
				})
```

Reply:
I love this idea!!! This will make the tests so much simpler & easier to understand!

My one nit: could we bake the privileges & image type into the name? So something like `some-readable-writeable-app-image` or `some-readable-cache-image` which I **think** would make the tests easier to follow.

Taking a look at the imgutil PR...